presto-spark-base/src/main/java/com/facebook/presto/spark/execution/http/PrestoSparkHttpWorkerClient.java

```diff
@@ -0,0 +1,115 @@
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_MAX_SIZE;
+import static java.util.Objects.requireNonNull;
+
+@ThreadSafe
+public class PrestoSparkHttpWorkerClient
+{
+    private static final Logger log = Logger.get(
+            PrestoSparkHttpWorkerClient.class);
+
+    private final HttpClient httpClient;
+    private final URI location;
+
+    public PrestoSparkHttpWorkerClient(HttpClient httpClient, URI location)
+    {
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+        this.location = requireNonNull(location, "location is null");
+    }
+
+    public ListenableFuture<PagesResponse> getResults(TaskId taskId, long token,
+            DataSize maxResponseSize)
+    {
+        URI uri = uriBuilderFrom(location)
+                .appendPath(taskId.toString())
+                .appendPath("/results/0")
+                .appendPath(String.valueOf(token))
+                .build();
+        return httpClient.executeAsync(
+                prepareGet()
+                        .setHeader(PRESTO_MAX_SIZE, maxResponseSize.toString())
+                        .setUri(uri).build(),
+                new PageResponseHandler());
+    }
+
+    public void acknowledgeResultsAsync(TaskId taskId, long nextToken)
+    {
+        URI uri = uriBuilderFrom(location)
+                .appendPath(taskId.toString())
+                .appendPath("/results/0")
+                .appendPath(String.valueOf(nextToken))
+                .appendPath("acknowledge")
+                .build();
+        httpClient.executeAsync(prepareGet().setUri(uri).build(),
+                new ResponseHandler<Void, RuntimeException>()
+                {
+                    @Override
+                    public Void handleException(Request request,
+                            Exception exception)
+                    {
+                        log.debug(exception, "Acknowledge request failed: %s",
+                                uri);
+                        return null;
+                    }
+
+                    @Override
+                    public Void handle(Request request, Response response)
+                    {
+                        if (familyForStatusCode(
+                                response.getStatusCode()) != HttpStatus.Family.SUCCESSFUL) {
+                            log.debug(
+                                    "Unexpected acknowledge response code: %s",
+                                    response.getStatusCode());
+                        }
+                        return null;
+                    }
+                });
+    }
+
+    public ListenableFuture<?> abortResults(TaskId taskId)
+    {
+        URI uri = uriBuilderFrom(location)
+                .appendPath(taskId.toString())
+                .build();
+        return httpClient.executeAsync(prepareDelete().setUri(uri).build(),
+                createStatusResponseHandler());
+    }
+
+    public URI getLocation()
```

Comment:
not used

Reply:
This is used in composing error message in HttpRemoteTaskResultFetcherRunner